### PR TITLE
Rewrite `extern-flag-rename-transitive`. `debugger-visualizer-dep-info`, `metadata-flag-frobs-symbols`, `extern-overrides-distribution` and `forced-unwind-terminate-pof` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

try-job: dist-x86_64-apple

### CODE REVIEW

**Comment by Oneirical** [comment]
Did you r- by muscle memory? Hehe

So, it seems like this `forced-unwind-terminate-pof` test uses `pthread`, a POSIX exclusive. I nerfed the ignore down from `only-linux` to `ignore-windows`, as it should work on MacOS/BSD too.

**Comment by jieyouxu** [comment]
> Did you r- by muscle memory? Hehe

Sorry, I was doing rollups at the time and was doing r- for a couple of different PRs and saw test failed in this PR lol

**Comment by jieyouxu** [comment]
Yes, tests look good but let's run a try job to double check

@bors try

**Comment by jieyouxu** [comment]
r=me if the try job comes back green

File: tests/run-make/metadata-flag-frobs-symbols/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+fn main() {
+    rustc().input("foo.rs").metadata("a").extra_filename("-a").run();
+    rustc().input("foo.rs").metadata("b").extra_filename("-b").run();
+    rustc()
+        .input("bar.rs")
+        .extern_("foo1", rust_lib_name("foo-a"))
+        .extern_("foo2", rust_lib_name("foo-b"))
+        .print("link-args")
+        .run();
```
Comment (jieyouxu) [inline tests/run-make/metadata-flag-frobs-symbols/rmake.rs:19]: Question: is this supposed to be checking otuput of `--print=link-args`? I know the original test does the same thing.

Comment (Oneirical) [reply tests/run-make/metadata-flag-frobs-symbols/rmake.rs:19]: You mean, if it should be `run_unchecked`? I don't see why, if the original test worked on all platforms as `run`, why would we need to reduce the exactitude of the test by not checking?

Comment (jieyouxu) [reply tests/run-make/metadata-flag-frobs-symbols/rmake.rs:19]: No I mean that this invocation uses `--print=link-args` but we're not checking what stdout contains (or what the content of `--print=link-args` is), is that intentional?

Comment (Oneirical) [reply tests/run-make/metadata-flag-frobs-symbols/rmake.rs:19]: That is quite weird indeed, my guess is that the test writer left it in as a way to debug their own test if it failed. The test works locally without it, so I am removing it now.

Comment (jieyouxu) [reply tests/run-make/metadata-flag-frobs-symbols/rmake.rs]: Thanks, r=me when CI passes

**Pull request merged**